compiler/luci/service/src/Nodes/CirclePad.test.cpp

```diff
@@ -90,3 +90,56 @@ TEST(ShapeRuleTest, pad_without_padding_NEG)
 
   ASSERT_ANY_THROW(shape_inf_rule.infer(node_pad, shape));
 }
+
+TEST(ShapeRuleTest, pad_non_const_paddings)
+{
+  auto g = loco::make_graph();
+  auto node_pad = g->nodes()->create<luci::CirclePad>();
+
+  auto node_paddings = g->nodes()->create<luci::CircleInput>();
+  auto node_input = g->nodes()->create<luci::CircleInput>();
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  node_input->shape({1, 2, 3, 4});
+  node_input->shape_status(luci::ShapeStatus::VALID);
+
+  node_paddings->dtype(loco::DataType::S64);
+  node_paddings->shape({4, 2});
+  node_paddings->shape_status(luci::ShapeStatus::VALID);
+
+  node_pad->input(node_input);
+  node_pad->paddings(node_paddings);
+
+  ASSERT_TRUE(shape_inf_rule.infer(node_pad, shape));
+  ASSERT_EQ(shape.rank(), 4);
+  ASSERT_FALSE(shape.dim(0).known());
+  ASSERT_FALSE(shape.dim(1).known());
+  ASSERT_FALSE(shape.dim(2).known());
+  ASSERT_FALSE(shape.dim(3).known());
+
+  ASSERT_EQ(0, shape.dim(0).value());
+  ASSERT_EQ(0, shape.dim(1).value());
+  ASSERT_EQ(0, shape.dim(2).value());
+  ASSERT_EQ(0, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, pad_empty_padding_NEG)
```

Comment:
You need to change the name of this test.

```suggestion
TEST(ShapeRuleTest, pad_non_const_node_padding_NEG)
```

Reply:
@seanshpark 

What do you think about this new `NEG` test?

Reply:
Please discuss with @shs-park and @zetwhite :)

Reply:
copy-paste from https://github.com/Samsung/ONE/pull/13875#discussion_r1751156686 

I had talk with @seanshpark and @shs-park. 
We have reached a consensus not to add negative tests for `LUCI_ASSERT`.

Because... 
  * For consistency, `LUCI_ASSERT` in another files are not checked in negative test. 
  * Since `assert` works differently in `release` and `debug` build, it is a bit troublesome to write the code for `assert`. 

In short, you could remove `assert` related NEG tests. 

-----------------------------------------------

BTW, Since we should have more negative tests than positive tests (by SE policy), we need to find another place(maybe `luci/imporeter`..?) to add Neg tests. Adding another NEG tests can be work in another PRs, not in this PR. -> Let's talk about this, by making new issue.

Reply:
@shs-park, please let me know if I misunderstood things or if i missed sth. :)